### PR TITLE
Enhance performance results output

### PR DESCRIPTION
**Pull request by safiranugroho**
✨ The CLI now:
- Expects two directories: one containing test results for current state, and one containing the baseline results to compare it against.
- Outputs the following files:
   - Summary of the test results, one file per directory
   - Summary of calculated difference between the two result sets, presented as a percent, along with the raw results from either directories consolidated into one file
   - Summary of the above results in ADF ready for import into Confluence

import-adf-low

### CODE REVIEW

File: src/cli/adf.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  const outputTables = Object.entries(calculationsByGroupId)
+    .map(([key, results]) => buildTable(key, results.map(buildTableRows)))
+    .flatMap((table) => table);
+
+  const adf = buildAdf(outputTables);
+
+  /**
+   * Write the generated ADF into file.
+   */
+  writeFile(ProcessDescription.ADF, 'adf', JSON.stringify(adf));
```
Comment (DarkPurple141) [inline src/cli/adf.ts:20]: this call is implicitly async, which is okay I guess although it does make the execution slightly less obvious

File: src/cli/calculate.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import {
```
Comment (DarkPurple141) [inline src/cli/calculate.ts:1]: nit: import type

File: src/cli/calculate.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import { performAllCalculations } from './util/calculate';
+import { writeFile } from './util/write';
+
+const calculate = (resultsByType: (ResultsByGroupId & { type: ResultType })[]) => {
+  /**
+   * Calculate the max, min, median, and mean values
+   * of the previously formatted outputs, and group the results
+   * per result type (baseline vs. current).
+   */
+  const calculationsByResultType = resultsByType.reduce(
```
Comment (DarkPurple141) [inline src/cli/calculate.ts:17]: I've read recently, and am now convinced that reducing with spreads is not ideal, and maybe a Object.fromEntries() is a better way to go about the same thing (better perf, easier to follow). Mind here it's a bit of a nit, but up to you if you think the change makes ssense.

File: src/cli/main.ts [hunk 4]
```diff
@@ -1,80 +1,88 @@
...
+    return usage();
+  }
 
-  const resultNames: string[] = [];
-  const resultSets: Row[][] = [];
+  const resultsByType = Object.entries(inputPaths).map(([type, pathName]) => {
+    const files = fs.readdirSync(pathName, 'utf-8');
+    if (!files) {
+      return debug(
+        `💔 Directory '${pathName}' is empty - ` +
```
Comment (DarkPurple141) [inline src/cli/main.ts:40]: <3 these messages

File: src/cli/utils.ts [hunk 5]
```diff
@@ -1,118 +0,0 @@
...
-  rows.forEach(({ key, samples, minValue, maxValue, meanValue, medianValue }) => {
-    stdout(`${key},${samples},${minValue},${maxValue},${meanValue},${medianValue}`);
-  });
-};
-
-const padded = (padding: number) => (str: string | number) => `${str}`.padEnd(padding);
-const paddedKey = padded(50);
-const paddedValue = padded(20);
-
-export const printCSVSummary = (resultNames: string[], results: Row[][]) => {
```
Comment (DarkPurple141) [inline src/cli/utils.ts:100]: I'm just wondering if we should keep this for glance value or does it complicate things?

**Review by DarkPurple141 — APPROVED** [review]
Cool, this is great. Mostly just keen to get in but maybe you might have some thoughts too.

File: src/cli/utils.ts (hunk 5 above)
Comment (safiranugroho) [reply src/cli/utils.ts:100]: hmm i feel like it has less value if the comparison data is not there though? maybe we'll bring it back if we want to iterate on it and include the comparison data?

File: src/cli/calculate.ts (hunk 3 above)
Comment (safiranugroho) [reply src/cli/calculate.ts:17]: nice love it

File: src/cli/calculate.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      .map(([type, resultByGroupId]) => [
+        type,
+        // This is where we actually perform the calculations, the rest of the mapping is
+        // purely to dig deep into the structure so we can extract the result out since they're grouped first
+        // by type (current vs. baseline) then by group id ("Server", "Client", "Interactions").
+        resultByGroupId.map(([groupId, result]) => [groupId, performCalculations(result)]),
+      ])
+      .map((entry) => entry as [ResultType, [string, Calculation[]][]])
+      .map(([type, calculationsByGroupId]) => [type, Object.fromEntries(calculationsByGroupId)]),
+  );
```
Comment (safiranugroho) [inline src/cli/calculate.ts:26]: @DarkPurple141 like this?

**Pull request merged**